ocaml/xapi/config_file_sync.ml

```diff
@@ -48,23 +48,6 @@ let write_to_fd s msg =
 let transmit_config_files s =
   read_config_file () |> rpc_of_config |> Jsonrpc.to_string |> write_to_fd s
 
-(* We still need to respect older XenServer hosts which are expecting
-   the entire /etc/password file. We need to make sure we send the
-   "un-shadowed" passwd file, so that slaves don't overwrite root's
-   password with an 'x'. *)
-
-(* This was introduced in 8c4d3d93f8c0e069b8b0de9c2e2df02a391aa2ef xapi
- * version 1.9.41 and can be removed when during rolling pool
- * upgrade the config_sync_version is always passed from slave to
- * master. We believe that Dundee is the first version where this code
- * shipped. Hence, it can be removed once RPU from pre-Dundee is no
- * longer supported.
- *)
-
-let legacy_transmit_passwd s =
-  debug "Updating /etc/passwd (%s)" __LOC__ ;
-  Unixpwd.unshadow () |> write_to_fd s
```

Comment:
We can remove this function from `unixpwd/src/unixpwd.mli` (and its tests, and maybe also some function from its c stubs)

Reply:
Can't quite figure out what uses `unixpwd/c/main.c` - everything (`make && make check && make test`) seems to be working alright even if I add syntax errors to it. Seems like only the local `unixpwd/c/Makefile` builds it - do we ever do this?

Reply:
If it's not installed, it was probably used for testing during development. Delete it. If we need it, we can make an ocaml-based executable.

Reply:
Removed these.